vip-rest-api.php

```diff
@@ -124,6 +121,32 @@ public function list_sites() {
 
 		return new WP_REST_Response( $sites );
 	}
+
+	/**
+	 * Check if necessary authentication header is present
+	 *
+	 * @return bool
+	 */
+	public function request_allowed( $request ) {
+		$header = $request->get_header( 'X-WPCom-Request-Secret' );
```

Comment:
I think we should use the standard `Authorization` header with a custom scheme instead of a custom header altogether (see http://stackoverflow.com/a/11420667). Maybe something like `Authorization: VIP-MACHINE-TOKEN a1b2c3d4...`